convert2rhel/unit_tests/main_test.py

```diff
@@ -68,6 +70,76 @@ def test_backup_control_unknown_exception(self, monkeypatch, global_backup_contr
         with pytest.raises(IndexError, match="Raised because of a bug in the code"):
             main.rollback_changes()
 
+    @pytest.mark.parametrize(
+        ("pre_conversion_results", "include_all_reports", "rollback_failures", "message", "summary_call_count"),
+        (
+            (
+                "anything",
+                False,
+                ["rollback_fail_0", "rollback_fail_1", "rollback_fail_2", "rollback_fail_3"],
+                "Rollback of system wasn't completed successfully.\n"
+                "The system is left in an undetermined state that Convert2RHEL cannot fix.\n"
+                "It is strongly recommended to store the Convert2RHEL logs for later investigation, and restore"
+                " the system from a backup.\n"
+                "Following errors were captured during rollback:\n"
+                "rollback_fail_0\n"
+                "rollback_fail_1\n"
+                "rollback_fail_2\n"
+                "rollback_fail_3",
+                0,
+            ),
+            (
+                "anything",
+                False,
+                ["rollback_fail_0"],
+                "Rollback of system wasn't completed successfully.\n"
+                "The system is left in an undetermined state that Convert2RHEL cannot fix.\n"
+                "It is strongly recommended to store the Convert2RHEL logs for later investigation, and restore"
+                " the system from a backup.\n"
+                "Following errors were captured during rollback:\n"
+                "rollback_fail_0",
+                0,
+            ),
+            ("anything", False, [], "", 1),
+            (
+                None,
+                True,
+                [],
+                "\nConversion interrupted before analysis of system completed. Report not generated.\n",
+                0,
+            ),
+        ),
+    )
+    def test_provide_status_after_rollback(
+        self,
+        monkeypatch,
+        caplog,
+        pre_conversion_results,
+        include_all_reports,
+        message,
+        global_backup_control,
+        rollback_failures,
+        summary_call_count,
+    ):
+        report_summary = mock.Mock()
+        monkeypatch.setattr(global_backup_control, "_rollback_failures", rollback_failures)
+        monkeypatch.setattr(report, "summary", report_summary)
+
+        main.provide_status_after_rollback(pre_conversion_results, include_all_reports)
+
+        assert report_summary.call_count == summary_call_count
+        if summary_call_count:
+            report_summary.assert_called_with(
+                results=pre_conversion_results,
+                include_all_reports=include_all_reports,
+                disable_colors=logger_module.should_disable_color_output(),
+            )
+        try:
+            assert message == caplog.records[-1].message
+        except IndexError:
+            # Nothing printed
+            assert message == caplog.text
```

Comment:
This call only tests the logs atm. Since the other calls, particularly `summary()` is mocked out it is not being tested correctly, instead we're testing a mock

If you can, remove the mock for report and mock out the file handling things instead

Reply:
@SpyTec I was thinking about it like the `summary()` is being tested elsewhere and here I just need to test if the appropriate log is printed in the right moment and also report is generated when needed (and called with right values). 

But I think I can change that, it won't be a big problem I believe.

Reply:
Testing logs is fine but we're supposed to test that the function works, not just that it calls what we want. Better to mock out files IMO